#! /usr/bin/env python
PACKAGE = "crazyflie_controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Enable trajectory tracking
gen.add("enable_traj_tracking", bool_t, 2, "Enable trajectory tracking for NMPC", False)

# Enable regulation for hovering at a certain position
gen.add("enable_regulation", bool_t, 2, "Enable regulation for NMPC", True)

# Regulation point
gen.add("xq_des", double_t, 64, "Desired position along the X axis", 0.0, -1.0, 1.0)
gen.add("yq_des", double_t, 64, "Desired position along the Y axis", 0.0, -1.0, 1.0)
gen.add("zq_des", double_t, 64, "Desired position along the Z axis", 0.40, 0.0, 1.0)

gen.add("delay", double_t, 64, "Compensated delay", 0.0, 0.015, 0.30)



exit(gen.generate(PACKAGE, "crazyflie_controller", "crazyflie_params"))
